vagrant/ansible/roles/samba-glusterfs.setup/tasks/main.yml

```diff
@@ -21,52 +21,48 @@
     state: yes
     persistent: yes
 
+- name: Create directory /etc/samba/fragments
+  file:
+    path: /etc/samba/fragments
+    state: directory
+    mode: '0755'
+
 - name: Copy over smb.conf
   template:
     src: smb.conf.j2
-    dest: /etc/samba/smb.conf
+    dest: /etc/samba/fragments/1smb.conf
     owner: root
     group: root
     mode: '0644'
 
-- name: Set volume options
-  gluster_volume:
-    state: present
-    name: "{{ gluster_cluster_volume }}"
-    options:
-      {
-        features.cache-invalidation: 'on',
-        features.cache-invalidation-timeout: '600',
-        performance.cache-samba-metadata: 'on',
-        performance.stat-prefetch: 'on',
-        performance.cache-invalidation: 'on',
-        performance.md-cache-timeout: '600',
-        network.inode-lru-limit: '200000',
-        performance.nl-cache: 'on',
-        performance.nl-cache-timeout: '600',
-        performance.readdir-ahead: 'on',
-        performance.parallel-readdir: 'on',
-      }
-  run_once: true
-
-- name: Mount glusterfs using fuse mount and set world writable permissions
+- name: Add share configuration
   block:
-    - name: Mount gluster volume using glusterfs
-      mount:
-        path: /mnt-fuse
-        src: 'localhost:/vol'
-        fstype: glusterfs
-        state: mounted
+    - name: Create temp file
+      template:
+        src: smb_share.conf.j2
+        dest: "/etc/samba/fragments/{{ samba_share }}.conf"
+      vars:
+        cluster_volume: "{{ item.cluster_volume }}"
+        samba_share: "{{ item.share_name }}"
+      with_items: "{{ samba_shares }}"
+
 
-    - name: Set permission of share root to 0777
-      file:
-        path: /mnt-fuse
-        mode: '0777'
+- name: Create /etc/samba/smb.conf
+  assemble:
+    src: /etc/samba/fragments
+    dest: /etc/samba/smb.conf
```

Comment:
How about using _include_ parameter? In that case we don't even need an additional directory.

Reply:
done.